worker/lease/manifold/manifold.go

```diff
@@ -102,9 +103,12 @@ func (s *manifoldState) start(context dependency.Context) (worker.Worker, error)
 		return nil, errors.Trace(err)
 	}
 
+	dbCtx, cancelCtx := context.WithCancel(context.Background())
```

Comment:
Shouldn't Manifold.start() have *some* sort of tomb/context/etc to indicate that the object you are starting has a piece for teardown?
(what you've done is fine, but I'm wondering why this isn't already entangled with a teardown because we are stopping).
Or is this actually part of 'NewWorker' and we are lacking a way to set it up early?